code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+	var/spider_debug = 0
+
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: SHARED TARGETING & ATTACK CODE -----------
+// --------------------------------------------------------------------------------
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/ListTargets()
+	var/list/targets1 = list()
+	var/list/targets2 = list()
+	var/list/targets3 = list()
+	if (ai_type == 0)
+		// default, BE AGGRESSIVE
+		//var/list/Mobs = hearers(vision_range, src) - src // this is how ListTargets for /mob/living/simple_animal/hostile/ does it, but it is wrong, it ignores NPCs.
+		for(var/mob/living/H in view(src, vision_range))
+		//for(var/mob/H in Mobs)
+			if (H.stat == 2)
+				// dead mobs are ALWAYS ignored.
+			else if (H.flags & GODMODE)
+				// if it is literally invincible, ignore it.
+			else if (!stat_attack && H.stat == 1)
+				// unconscious mobs are ignored unless spider has stat_attack
+			else if (istype(H, /mob/living/simple_animal/hostile/poison/terror_spider))
+				// fellow terror spiders are never valid targets unless they deliberately attack us, even then, low priority
+				if (H in enemies)
+					targets3 += H
+			else if (H.reagents)
+				if (H.paralysis && H.reagents.has_reagent("terror_white_tranq"))
+					// let's not target completely paralysed mobs.
+					if (H in enemies)
+						targets3 += H
+						// unless we hate their guts
+				if (IsInfected(H)) // target them if they attack us
+					if (H in enemies)
+						targets3 += H
+				else if (H.reagents.has_reagent("terror_black_toxin") && istype(src, /mob/living/simple_animal/hostile/poison/terror_spider/black))
+					if (get_dist(src,H) <= 2)
+						// if they come to us...
+						targets2 += H
+					else if ((H in enemies) && !H.reagents.has_reagent("terror_black_toxin",31))
+						// if we're aggressive, and they're not going to die quickly...
+						targets2 += H
+					else
+						// they are far away, and either we're not very aggressive, or they are dying already
+						// either way, not much point in targeting them
+						// if they shoot us, of course, then we will consider them a valid target
+				else
+					if (H.can_inject(null,0,"chest",0))
+						targets1 += H
+					else if (H in enemies)
+						targets2 += H
+					else
+						targets3 += H
+					// first, go after targets we can inject - lightly armored humans.
+					// second, go after humans that attacked us.
+					// third, go after other humans.
+			else if (istype(H, /mob/living/simple_animal))
+				var/mob/living/simple_animal/hostile/poison/terror_spider/M = H
+				if (M.force_threshold > melee_damage_upper)
+					// If it has such high armor it can ignore any attack we make on it, ignore it.
+				else if (M in enemies)
+					targets2 += M
+				else
+					targets3 += M
+			else
+				if (H in enemies)
+					targets2 += H
+				else
+					targets3 += H
+		for(var/obj/mecha/M in mechas_list)
```

Comment:
Because that is how https://github.com/ParadiseSS13/Paradise/blob/64dbdc42db5a6f7f5fb674e39d7463cf5979a253/code/modules/mob/living/simple_animal/hostile/hostile.dm @ line 69 does it, and I assumed it was the standard way of doing things, because that's how all simple_animals apparently handle mechs.